### ISSUE TITLE
Include "type signature" from the schema.yml docs in the ref preview

### ISSUE THREAD

**Issue by jsnb-devoted** [AltimateAI/vscode-dbt-power-user#525]
### Describe the feature

The current command+hover preview looks like:
![Screenshot 2023-08-14 at 2 14 15 PM](https://github.com/innoverio/vscode-dbt-power-user/assets/79863718/19347b47-4745-47ef-be03-41a552e4c735)

It includes the config block from the model. It would be nice to see some more metadata in there like the table description or even the columns if it could be presented in a readable way.

### Describe alternatives you've considered

You can kind of get to this with an extra click if you go-to the definition of the ref and read the docs but then you are opening a new file.

### Who will benefit?

This is a pretty generalizable feature. I suspect that most folks don't use the preview as much as they just click through to the definition.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by anandgupta42** [comment]
@jsnb-devoted Thanks for creating the issues; looking forward to the PR. Let us know if we can help.

**Comment by jsnb-devoted** [comment]
@anandgupta42 -- I started looking into this and I found the vscode api docs hard to follow. I've gone down a few paths and I'm not sure if any of them are correct:
1. Amend the existing `registerDefinitionProvider` to augment the command+hover behavior to show metadata about the symbol
2. Add the [registerTypeDefinitionProvider](https://code.visualstudio.com/api/references/vscode-api#:~:text=registerTypeDefinitionProvider) (or maybe replace the `registerDefinitionProvider`??) that will definite "types" for the symbols.
3. Totally separate from 1+2 -- create a [registerHoverProvider](https://code.visualstudio.com/api/references/vscode-api#:~:text=)%3A%20Disposable-,registerHoverProvider,-(selector%3A) that shows type metadata when you hover (without the command button).

I am leaning towards #3 being the option here. It seems like the hover capability is actually what we want for this. I just thought there was something a little more formal for displaying "type" information.

If any folks have any experience with this and can save me some time doing trial+error and spelunking thru the docs that would be a great help.

**Comment by mdesmet** [comment]
To be totally clear, I haven't actually tried out the HoverProvider myself but based on the docs it seems indeed like the way to go here.

By looking at the interface of the `HoverProvider` it seems to work very similar to the `DefinitionProvider`. I think you can try to follow a same path for implementation.

Let me know how it goes!

**Comment by jsnb-devoted** [comment]
Copy/pasting my way into making a little progress on this:
<img width="648" alt="Screenshot 2023-09-16 at 12 15 23 PM" src="https://github.com/AltimateAI/vscode-dbt-power-user/assets/79863718/9bf53138-3f63-4bc2-9fd2-9ef7e9056c7c">
